client/src/main/java/io/confluent/kafka/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -25,15 +25,15 @@
 
   public int register(String subject, Schema schema) throws IOException, RestClientException;
 
-  public Schema getByID(int id) throws IOException, RestClientException;
+  public Schema getBySubjectAndID(String subject, int id) throws IOException, RestClientException;
```

Comment:
Strictly speaking this isn't publicly supported API, but I think some people may be using this interface and the cached client directly in their own Java code. Could we leave the `getByID` version in and just have the trivial implementation in each of the implementations (i.e. they will just call `getBySubjectAndID(null, id)`)?